AEFeinstein/Super-2024-Swadge-FW
# Ultimate TTT AI

### Description

Adds a single-player CPU opponent with 3 difficulties:

* Easy -- Plays a random move
* Medium -- Plays optimally within sub-games, but has no overall strategy
* Hard -- Analyzes the board and tries to play the best overall move, taking into account the best sub-game move, and weighing that against what the opponent's best move would be in the sub-game they would get to move in. Or, if they would get to pick any sub-game, their best move overall.

Also changes `tttHandleGameInput()` to use a slightly different method, which should prevent any squares from becoming inaccessible and I think maybe it handles a bit more intuitively than the previous one? I recommend playing with it to see. But it pretty much tries to find a close free space before it wraps all the way around by bumping the cursor away from the edge of the board, and it also prioritizes the last intended direction for the secondary cursor direction.

### Test Instructions

Play against the AIs, see how they feel. See if the cursor behaves reasonably in some funky situations.

### Ticket Links

Closes #267 

### Readiness Checklist
